segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -157,20 +157,22 @@ public void setupAppend(SetupAppend setupAppend) {
 
         // Get the last Event Number for this writer from the Store. This operation (cache=true) will automatically put
         // the value in the Store's cache so it's faster to access later.
-        store.getAttributes(newSegment, Collections.singleton(writer), true, TIMEOUT)
-                .whenComplete((attributes, u) -> {
-                    try {
-                        if (u != null) {
-                            handleException(writer, setupAppend.getRequestId(), newSegment, "setting up append", u);
-                        } else {
-                            long eventNumber = attributes.getOrDefault(writer, Attributes.NULL_ATTRIBUTE_VALUE);
-                            this.writerStates.putIfAbsent(Pair.of(newSegment, writer), new WriterState(eventNumber));
-                            connection.send(new AppendSetup(setupAppend.getRequestId(), newSegment, writer, eventNumber));
-                        }
-                    } catch (Throwable e) {
-                        handleException(writer, setupAppend.getRequestId(), newSegment, "handling setupAppend result", e);
-                    }
-                });
+        Futures.exceptionallyComposeExpecting(
+                store.getAttributes(newSegment, Collections.singleton(writer), true, TIMEOUT),
+                e -> e instanceof StreamSegmentSealedException, () -> store.getAttributes(newSegment, Collections.singleton(writer), false, TIMEOUT))
```

Comment:
This looks good, however I want some way of logging that we are indeed running into some sealed segment. At the same time, I do not want to pollute the logs with useless messages. Looking through this class, it seems that a subsequent append via this connection-writer will stumble across a sealed segment which will take care of some logging and cleanup (OK - nothing to do there).

I just took a look at the `getAttributes` call inside the Segment Store and it turns out we are not logging the `cache` argument. Would you mind editing line 933 in `StreamSegmentContainer.java` and add the `cache` parameter to that `logRequest` call please? That way we will be able to see the first and the second call in the logs with proper args.